service/src/main/java/org/cbioportal/service/impl/MolecularDataServiceImpl.java

```diff
@@ -222,4 +224,64 @@ private void validateMolecularProfile(String molecularProfileId) throws Molecula
             throw new MolecularProfileNotFoundException(molecularProfileId);
         }
     }
+
+    @Override
+    public List<GenesetMolecularData> getGenesetMolecularData(String molecularProfileId, String sampleListId,
+            List<String> genesetIds, String projection) throws MolecularProfileNotFoundException {
+        validateMolecularProfile(molecularProfileId);
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return fetchGenesetMolecularData(molecularProfileId, sampleIds, genesetIds, projection);
+    }
+    
+    @Override
+    public List<GenesetMolecularData> fetchGenesetMolecularData(String molecularProfileId, List<String> sampleIds,
```

Comment:
This is still pretty much copy paste of `fetchMolecularData` except one line. Instead of doing that, we should change `molecularDataRepository.getGeneMolecularAlterations` to accept both genes and genesets and call the appropriate mapper method based on the genetic entity type.

Reply:
@ersinciftci My fault, there was already an available endpoint to fetch molecular data for gene sets.